src/filtered-list/index.ts

```diff
@@ -23,6 +25,17 @@ class FilteredList {
     this.updateItems();
   }
 
+  onKeyDown(event, value: string = this.refs.filter.value) {
+    value = value.trim().toLowerCase();
+    if (event.keyCode === RETURN_KEY_CODE) {
+      const foundItem = this.props.items.find((item) => typeof item === 'object' && item.value && item.value.toLowerCase() === value);
```

Comment:
Don't check if item is an `object`, because in this case if item is `null` this code will error. It sounds like you're having trouble with type guards. What ts docs say to do here (https://www.typescriptlang.org/docs/handbook/advanced-types.html search for type guard on this page) is to cast it (in other words, you need to pull the object type out into a separate type in the `Item` type definition below, then here cast the type as needed.